woocommerce-gutenberg-products-block.php

```diff
@@ -261,3 +261,54 @@ function woocommerce_blocks_get_php_translation_from_core( $translation, $text,
 }
 
 add_filter( 'gettext', 'woocommerce_blocks_get_php_translation_from_core', 10, 3 );
+
+/**
+ * Check the current WC Blocks path. If the WC Blocks plugin is active but the
+ * current path is from the WC Core, we can consider the plugin is outdated
+ * because JetPack autoloader always loads the newer package.
+ *
+ * @see https://github.com/woocommerce/woocommerce-blocks/issues/5587
+ *
+ * @return bool
+ */
+function woocommerce_blocks_plugin_is_outdated() {
+	$is_active      = false;
+	$active_plugins = get_option( 'active_plugins', array() );
+
+	// We check for the main plugin file only instead of the full path, because
+	// the plugin folder may not be 'woocommerce-blocks'.
+	foreach ( $active_plugins as $plugin ) {
+		if ( strpos( $plugin, 'woocommerce-gutenberg-products-block.php' ) ) {
+			$is_active = true;
+			break;
+		}
+	}
```

Comment:
I don't have a strong opinion, but we could alternatively use [`is_plugin_active()`](https://developer.wordpress.org/reference/functions/is_plugin_active/) here, right? Did you discard it for some reason?
```PHP
$is_active =
	is_plugin_active( 'woocommerce-gutenberg-products-block/woocommerce-gutenberg-products-block.php' ) ||
	is_plugin_active( 'woocommerce-blocks/woocommerce-gutenberg-products-block.php' );
```

(I tested and that function is not available yet when this code is executed, but we could easily move the `woocommerce_blocks_plugin_is_outdated()` check inside the `woocommerce_blocks_plugin_outdated_notice()` function to solve that)

Reply:
@Aljullu yes, we can use `is_plugin_active()`. I did it the current way so the code works with any plugin folder name. We already have three possible names:
- `woo-gutenberg-products-block` (which is used by the .org plugin).
- `woocommerce-blocks`.
- `woocommerce-gutenberg-products-block`.

Reply:
After considering, I decided to use `is_plugin_active` and explicitly check for three possible cases.